src/service/extension-location.js

```diff
@@ -68,16 +77,22 @@ export function calculateExtensionScriptUrl(
   opt_extensionVersion,
   opt_isLocalDev
 ) {
-  const base = calculateScriptBaseUrl(location, opt_isLocalDev);
-  const rtv = getMode().rtvVersion;
+  const useDefaultCdn =
+    isCacheModifiedExtension(extensionId) &&
+    !urls.cdnSupportsCacheModifiedExtensions;
+  const base = calculateScriptBaseUrl(location, opt_isLocalDev, useDefaultCdn);
   if (opt_extensionVersion == null) {
     opt_extensionVersion = '0.1';
   }
   const extensionVersion = opt_extensionVersion
     ? '-' + opt_extensionVersion
     : '';
   const spPath = getSinglePassExperimentPath();
-  return `${base}/rtv/${rtv}/${spPath}v0/${extensionId}${extensionVersion}.js`;
+  const extensionPath = `${spPath}v0/${extensionId}${extensionVersion}.js`;
+  if (urls.cdn !== urls.defaultCdn && !useDefaultCdn) {
+    return `${base}/${extensionPath}`;
```

Comment:
Why are we allowing this without RTV fingerprint? The fingerprint allows for easy longterm caching (important for the browser, not the cache server), where a shared URL forces frequent rechecks.

I think prioritizing the user's experience means that, if you want to self-host, you need to follow the RTV contract.

Reply:
Note, too, that this can cause an infinite reload loop if the loaded extension does not match the version loaded by v0.js. We automatically reload requesting `/rtv/${V0's RTV}/v0/${ext}.js`. If we can't specifically request an RTV, then we'll keep requesting `/v0/${ext}.js`, which has the incorrect version (because we started the reload loop), and continue looping.

Reply:
> Why are we allowing this without RTV fingerprint?

Please read conversation thread here and at https://github.com/ampproject/amp-toolbox/pull/517.

> Note, too, that this can cause an infinite reload loop if the loaded extension does not match the version loaded by v0.js.

I'll review the reload conditions this evening to see if changes are warranted here or there.

Reply:
> Note, too, that this can cause an infinite reload loop if the loaded extension does not match the version loaded by v0.js. We automatically reload requesting `/rtv/${V0's RTV}/v0/${ext}.js`.

Commit 6bbbd5d disables version locking when self-hosting. It's up to the cache maintainer to ensure version alignment between runtime and components.